### ISSUE TITLE
`poetry add requests-ip-rotator` does not work

### ISSUE THREAD

**Issue by kmille** [python-poetry/poetry#9903]
### Description

Hey,
what's wrong with the repository?

```
kmille@linbox:freetar vim pyproject.toml
kmille@linbox:freetar poetry add requests-ip
Using version ^1.0.14 for requests-ip-rotato

Updating dependencies (0.6s)^C
```

- https://pypi.org/project/requests-ip-rotator/
- https://github.com/Ge0rg3/requests-ip-rotator

Seems like the repo is missing something. I can not add it as a dependency. It just says "Updating dpendencies" but nothing works.

### Workarounds

?

### Poetry Installation Method

pipx

### Operating System

Arch

### Poetry Version

Poetry (version 1.8.4)

### Poetry Configuration

```bash session
kmille@linbox:freetar poetry config --list
cache-dir = "/home/kmille/.cache/pypoetry"
experimental.system-git-client = false
installer.max-workers = null
installer.modern-installation = true
installer.no-binary = null
installer.parallel = true
keyring.enabled = true
solver.lazy-wheel = true
virtualenvs.create = true
virtualenvs.in-project = null
virtualenvs.options.always-copy = false
virtualenvs.options.no-pip = false
virtualenvs.options.no-setuptools = false
virtualenvs.options.system-site-packages = false
virtualenvs.path = "{cache-dir}/virtualenvs"  # /home/kmille/.cache/pypoetry/virtualenvs
virtualenvs.prefer-active-python = false
virtualenvs.prompt = "{project_name}-py{python_version}"
warnings.export = true
```

### Poetry Runtime Logs

```bash session
will check it
```

**Comment by kmille** [comment]
Using `poetry -vvv add requests-ip-rotator` shows me the acutal problem. I think I can now solve the issue, but the UI in this case is not great:

```
Source (PyPI): 0 packages found for botocore >1.35.81,<1.36.0
Falling back to installed packages to discover metadata for botocore
Found 0 compatible packages for botocore
   1: fact: no versions of botocore match >1.35.81,<1.36.0
   1: conflict: no versions of botocore match >1.35.81,<1.36.0                 !!!!!!!!!!!!!!!!!!!!!!!!!!!
   1: ! botocore (>1.35.81,<1.36.0) is partially satisfied by not botocore (==1.35.81)
   1: ! which is caused by "botocore (1.35.81) depends on urllib3 (>=1.25.4,<1.27)"
   1: ! thus: botocore (>=1.35.81,<1.36.0) requires urllib3 (>=1.25.4,<1.27)
   1: fact: botocore (>=1.35.81,<1.36.0) requires urllib3 (>=1.25.4,<1.27)
   1: derived: not botocore (>=1.35.81,<1.36.0)
   1: derived: not boto3 (==1.35.81)
   2: selecting idna (3.10)
   2: selecting charset-normalizer (3.4.0)
   2: selecting attrs (24.2.0)
   2: selecting pyrepl (0.9.0)
   2: selecting ply (3.11)
   2: selecting markupsafe (3.0.2)
   2: fact: boto3 (1.35.80) depends on botocore (>=1.35.80,<1.36.0)
   2: fact: boto3 (1.35.80) depends on jmespath (>=0.7.1,<2.0.0)
   2: fact: boto3 (1.35.80) depends on s3transfer (>=0.10.0,<0.11.0)
   2: selecting boto3 (1.35.80)
   2: derived: s3transfer (>=0.10.0,<0.11.0)
   2: derived: jmespath (>=0.7.1,<2.0.0)
   2: derived: botocore (>=1.35.80,<1.36.0)
```

It does this in a infinity loop and does not print a warning without `-vvv`

```ini
kmille@linbox:freetar cat pyproject.toml            
[tool.poetry]
name = "freetar"
version = "0.10.0"
description = ""
authors = ["kmille <github@androidloves.me>"]
readme = "README.md"
repository = "https://github.com/kmille/freetar"
homepage = "https://github.com/kmille/freetar"


[tool.poetry.dependencies]
python = "^3.9"
requests = {extras = ["socks"], version = "^2.32.3"}
beautifulsoup4 = "^4.12.0"
flask = "^3.0.3"
waitress = "^2.1.2"
flask-minify = "^0.48"
flask-caching = "^2.3.0"


[tool.poetry.group.dev.dependencies]
pdbpp = "^0.10.3"

[build-system]
requires = ["poetry-core"]
build-backend = "poetry.core.masonry.api"

[tool.poetry.scripts]
freetar = 'freetar.backend:main'
```

**Comment by dimbleby** [comment]
Search "botocore" to find duplicates, please close

**Comment by kmille** [comment]
Can you please link it? Can't find it.

**Comment by abn** [comment]
Duplicate of #8823.
